CardPayments/src/main/java/com/paypal/android/cardpayments/CardClient.kt

```diff
@@ -39,18 +39,17 @@ class CardClient internal constructor(
      */
     var cardVaultListener: CardVaultListener? = null
 
+    private var approveOrderId: String? = null
     private val lifeCycleObserver = CardLifeCycleObserver(this)
 
     private val approveOrderExceptionHandler = CoreCoroutineExceptionHandler { error ->
-        notifyApproveOrderFailure(error)
+        notifyApproveOrderFailure(error, approveOrderId)
     }
 
```

Comment:
Oh I guess we need to do this in iOS as well.

Reply:
Yeah I wish we could not have it tbh but this is in line with iOS.

Reply:
We may re-visit one day to see if we really need it. For now it's easiest to maintain analytics parity.